gitpod-io/gitpod
# [agent-smith] Stop workspace through ws-manager

## Description

Agent smith stops a workspace by calling ws-manager (with the `IMMEDIATELY` policy to remove a grace period). This fixes workspaces stopped by agent smith being counted as failed workspace stops in metrics.

## Related Issue(s)

Fixes WKS-207

## How to test


Test in this preview env: https://wv-agent-s9c7faea0ad.preview.gitpod-dev.com/workspaces

Agent smith is configured to stop workspaces running `sleep`. Run `sleep 40s` and observe the workspace stopping.

## Documentation


#### Preview status

gitpod:summary

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`
</details>

/hold
